domain-models-runtime/src/test/java/org/folio/rest/persist/PostgresClientIT.java

```diff
@@ -2219,51 +2222,447 @@ public void selectDistinctOn(TestContext context) throws IOException {
+        context.assertEquals(1, resultInfo.getFacets().get(1).getFacetValues().get(0).getCount());
+        context.assertEquals(1, resultInfo.getFacets().get(1).getFacetValues().get(1).getCount());
+        context.assertEquals(1, resultInfo.getFacets().get(1).getFacetValues().get(2).getCount());
+        sr.handler(streamHandler -> objectCount.incrementAndGet());
+        sr.endHandler(x -> {
           context.assertEquals(3, objectCount.get());
           async.complete();
         });
+      });
     async.awaitSuccess();
   }
 
   @Test
-  public void streamGetDistinctOn(TestContext context) throws IOException {
+  public void streamGetWithFacetsZeroHits(TestContext context) throws IOException, FieldException {
     AtomicInteger objectCount = new AtomicInteger();
     Async async = context.async();
 
     final String tableDefiniton = "id UUID PRIMARY KEY , jsonb JSONB NOT NULL, distinct_test_field TEXT";
+    List<FacetField> facets = new ArrayList<FacetField>();
+    facets.add(new FacetField("jsonb->>'edition'"));
+    facets.add(new FacetField("jsonb->>'title'"));
+    createTableWithPoLines(context, MOCK_POLINES_TABLE, tableDefiniton);
+    CQLWrapper wrapper = new CQLWrapper(new CQL2PgJSON("jsonb"), "edition=Millenium edition");
+    postgresClient.streamGet(MOCK_POLINES_TABLE, Object.class, "jsonb", wrapper, true, null,
+      facets, asyncResult -> {
+        context.assertTrue(asyncResult.succeeded());
+        PostgresClientStreamResult sr = asyncResult.result();
+        ResultInfo resultInfo = sr.resultInto();
+        context.assertEquals(0, resultInfo.getTotalRecords());
+        context.assertEquals(0, resultInfo.getFacets().size());
+        sr.handler(streamHandler -> objectCount.incrementAndGet());
+        sr.endHandler(x -> {
+          context.assertEquals(0, objectCount.get());
+          async.complete();
+        });
+      });
+    async.awaitSuccess();
+  }
 
+  @Test
+  public void streamGetWithFacetsError(TestContext context) throws IOException, FieldException {
+    AtomicInteger objectCount = new AtomicInteger();
+    Async async = context.async();
+
+    final String tableDefiniton = "id UUID PRIMARY KEY , jsonb JSONB NOT NULL, distinct_test_field TEXT";
+    List<FacetField> badFacets = new ArrayList<FacetField>();
+    badFacets.add(new FacetField("'"));  // bad facet
     createTableWithPoLines(context, MOCK_POLINES_TABLE, tableDefiniton);
-    postgresClient.streamGet(MOCK_POLINES_TABLE, new Object(), "jsonb", null, false, "jsonb->>'edition'",
-        streamHandler -> objectCount.incrementAndGet(), asyncResult -> {
+    CQLWrapper wrapper = new CQLWrapper(new CQL2PgJSON("jsonb"), "edition=First edition");
+    postgresClient.streamGet(MOCK_POLINES_TABLE, Object.class, "jsonb", wrapper, true, null,
+      badFacets, asyncResult -> {
+        context.assertTrue(asyncResult.failed());
+        async.complete();
+      });
+    async.awaitSuccess();
```

Comment:
Can you remove `Async async` and use `context.asyncAssertFailure()` instead?
Can you either remove `objectCount` or check the value?